app/services/post_scraper.rb

```diff
@@ -125,18 +97,13 @@ def threaded_page_links
 
   def import_post_from_doc(doc)
     subject = @subject || doc.at_css('.entry .entry-title').text.strip
-    logger.info "Importing thread '#{subject}'"
+    Resque.logger.info "Importing thread '#{subject}'"
 
-    @post = Post.new
-    @post.board_id = @board_id
-    @post.section_id = @section_id
-    @post.subject = subject
-    @post.status = @status
-    @post.is_import = true
+    @post = Post.new(board_id: @board_id, section_id: @section_id, subject: subject, status: @status, is_import: true)
 
     # detect already imported
     # skip if it's a threaded import, unless a subject was given manually
-    if (@subject || !@threaded_import) && (subj_post = Post.find_by(subject: @post.subject, board_id: @board_id))
+    if (subject || !@threaded_import) && (subj_post = Post.find_by(subject: subject, board_id: @board_id))
```

Comment:
This first condition, `subject || !@threaded_import`, will always be true - either we were passed a manual subject or we fetched it from the page

Previously it was checking specifically that it was passed manually to do the disambiguation here, I guess so people can force a reimport for some reason? We should probably keep that, or if we're sure we don't need that, reformulate this condition to just always do the disambiguation

```suggestion
    if (@subject || !@threaded_import) && (subj_post = Post.find_by(subject: subject, board_id: @board_id))
```